allennlp/data/iterators/data_iterator.py

```diff
@@ -243,6 +243,8 @@ def _ensure_batch_is_sufficiently_small(self, batch_instances: Iterable[Instance
         padding_length = -1
         list_batch_instances = list(batch_instances)
         for instance in list_batch_instances:
+            if self.vocab is not None:
```

Comment:
can you just add a comment as to why we're doing indexing here when we also do it elsewhere?